frontend/src/hooks/@common/useCodeHighlight.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react';
+import Prism from 'prismjs';
+import 'prismjs/themes/prism.css';
+
+const useCodeHighlight = (htmlDOMString?: string) => {
+  useEffect(() => {
+    if (!htmlDOMString) return;
+    const importPrism = async () => {
+      const parser = new DOMParser();
+      const doc = parser.parseFromString(htmlDOMString, 'text/html');
+
+      const codeElements = doc.querySelectorAll('code[class^="language-"]');
+      const languages = Array.from(codeElements)
+        .map((element) => {
+          const match = element.className.match(/language-(\w+)/);
+          return match ? match[1] : null;
+        })
+        .filter(Boolean);
```

Comment:
이거 맞는 지 모르겠는는데 이런식으로 써볼 수 있을 것 같아요!

```suggestion
      const languages = Array.from(codeElements)
      .filter((element) => {
           return element.className.includes(/language-(\w+)/);                                           
      }).map(match => match[1])
```

Reply:
dom파싱 부분 없애고 [eadacf9](https://github.com/woowacourse-teams/2023-dong-gle/pull/394/commits/eadacf9a31bdfa73323ff7d90265d6f5d38e5329) 로 수정했습니다.